py-polars/polars/io/database.py

```diff
@@ -792,10 +803,21 @@ def _read_sql_connectorx(
 
 
 def _read_sql_adbc(
-    query: str, connection_uri: str, schema_overrides: SchemaDict | None
+    query: str,
+    connection_uri: str,
+    schema_overrides: SchemaDict | None,
+    execute_options: dict[str, Any] | None = None,
 ) -> DataFrame:
     with _open_adbc_connection(connection_uri) as conn, conn.cursor() as cursor:
-        cursor.execute(query)
+        if (
+            execute_options
+            and (params := execute_options.get("parameters")) is not None
+        ):
+            if isinstance(params, dict):
+                params = pa.Table.from_pydict({k: [v] for k, v in params.items()})
+                execute_options["parameters"] = params
```

Comment:
>  If that results in a bad user experience, that's up to the ADBC folks to address. 

Or we can fix it in a single line of code so that we don't pass that poor interface on to our own users. (I am sorely tempted to make an upstream patch to address this; it seems... kind of horrible 😕).

Reply:
I really don't think it's a good idea to assume that responsibility. Upstream patch sounds good 👍

Reply:
Done (not the upstream patch; will eyeball that at the weekend).